src/Service/ElasticsearchService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * @file
+ * Contains service to integrate with elasticsearch.
+ */
+
+namespace App\Service;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Class ElasticsearchService.
+ */
+class ElasticsearchService implements ElasticsearchServiceInterface
+{
+    private const SEARCH_SIZE = 100;
+
+    private $elasticsearchURL;
+    private $httpClient;
+
+    /**
+     * StatisticsExtractionService constructor.
+     *
+     * @param \Symfony\Contracts\HttpClient\HttpClientInterface $httpClient
+     *   The http client
+     * @param $boundElasticsearchURL
+     *   Url of elasticsearch instance
+     */
+    public function __construct(HttpClientInterface $httpClient, $boundElasticsearchURL)
+    {
+        $this->elasticsearchURL = $boundElasticsearchURL;
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * Get the records from the given date from Elasticsearch.
+     *
+     * @param \DateTime $date
+     *   Requested date
+     * @param string $message
+     *   The message to search for
+     *
+     * @return array
+     *   Array of logs for the given date
+     *
+     * @throws \Throwable
```

Comment:
Would it not be an exception that it throws?

https://www.php.net/manual/en/class.throwable.php

Reply:
Fixed